format/avro/testdata/quickstop-deflate.fqtest

```diff
@@ -0,0 +1,121 @@
+$ fq verbose quickstop-deflate.avro
+      |00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f|0123456789abcdef|.{}: quickstop-deflate.avro (avro_ocf) 0x0-0x5835.7 (22582)
+0x0000|4f 62 6a 01                                    |Obj.            |  magic: raw bits (valid) 0x0-0x3.7 (4)
+      |                                               |                |  meta[0:2]: 0x4-0x119.7 (278)
+      |                                               |                |    [0]{}: block 0x4-0x118.7 (277)
+0x0000|            04                                 |    .           |      count: 2 0x4-0x4.7 (1)
+      |                                               |                |      entries[0:2]: 0x5-0x118.7 (276)
+      |                                               |                |        [0]{}: entry 0x5-0x17.7 (19)
+0x0000|               14                              |     .          |          key_len: 10 0x5-0x5.7 (1)
+0x0000|                  61 76 72 6f 2e 63 6f 64 65 63|      avro.codec|          key: "avro.codec" 0x6-0xf.7 (10)
+0x0010|0e                                             |.               |          value_len: 7 0x10-0x10.7 (1)
+0x0010|   64 65 66 6c 61 74 65                        | deflate        |          value: "deflate" 0x11-0x17.7 (7)
```

Comment:
Is this deflate as in deflate compressed? if so you could probably uncompress and add as new buffer if that would be useful

Reply:
Yeah this is probably the best option. How would you take the decoded byte array and use it with a decode.D?

Reply:
If the uncompressed data is just opaque and will not be decoded further you can do something like this i think:

```go
br := d.FieldRawLen("compressed", ...)
bb := &bytes.Buffer{}
d.MustCopy(bb, flate.NewReader(bitio.NewIOReader(br)))
d.FieldRootBitBuf("uncompressed", bitio.NewBitReader(bb.Bytes(), -1))
```

There are some helpers for doing similar things but they all seem to decode the uncompressed data somehow, like `d.TryFieldReaderRangeFormat`, `d.FieldFormatBitBuf` and `d.FieldRoot*Array/Struct/Fn`. Maybe there should be some helper for this case also? ...also really need to clean those functions up, but is a bit tricky to generelize i've noticed :)

Reply:
ty, used the d.FieldArrayRootBitBufFn